fish_prompt.fish

```diff
@@ -359,7 +359,7 @@ function __bobthefish_path_segment -S -a segment_dir -d 'Display a shortened for
     switch "$segment_dir"
         case /
             set directory '/'
-        case "$HOME"
+        case (builtin realpath "$HOME")
```

Comment:
let's do `case "$HOME" (builtin realpath "$HOME")`. the string comparison should be enough cheaper than a filesystem lookup that it's probably worth doing.